tests/gauge_path_test.cpp

```diff
@@ -214,6 +235,140 @@ void gauge_force_test(bool compute_force = true)
   delete Mom_ref_milc;
 }
 
+void gauge_loop_test()
+{
+  int max_length = 6;
+
+  setVerbosityQuda(QUDA_VERBOSE, "", stdout);
+
+  QudaGaugeParam gauge_param = newQudaGaugeParam();
+  setWilsonGaugeParam(gauge_param);
+
+  gauge_param.gauge_order = gauge_order;
+  gauge_param.t_boundary = QUDA_PERIODIC_T;
+
+  setDims(gauge_param.X);
+
+  double *trace_loop_coeff_p = &trace_loop_coeff_d[0];
+  int num_paths = sizeof(trace_path) / sizeof(trace_path[0]);
+  int *trace_loop_length_p = &trace_loop_length[0];
+
+  // b/c we can't cast int*[6] -> int**
+  int **trace_path_p = new int *[num_paths];
+  for (int i = 0; i < num_paths; i++) {
+    trace_path_p[i] = new int[max_length];
+    for (int j = 0; j < trace_loop_length[i]; j++) trace_path_p[i][j] = trace_path[i][j];
+  }
+
+  quda::GaugeFieldParam param(gauge_param);
+  param.create = QUDA_NULL_FIELD_CREATE;
+  param.order = QUDA_QDP_GAUGE_ORDER;
+  auto U_qdp = new quda::cpuGaugeField(param);
```

Comment:
does this need to be a ptr?

Reply:
Fixed throughout the file, good call, will push momentarily

Reply:
https://github.com/lattice/quda/pull/1308/commits/ccb7dd9dfe1dbb485f0d2f05456c9276a79fbafb